src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/WebInputSources.cs

```diff
@@ -12,6 +16,18 @@ internal static class WebInputSources
         /// </summary>
         public static ImmutableHashSet<SourceInfo> SourceInfos { get; }
 
+        /// <summary>
+        /// Cached information if the specified symbol is a Asp.Net Core Controller: (compilation) -> ((class symbol) -> (is Controller))
+        /// </summary>
+        private static readonly BoundedCacheWithFactory<Compilation, ConcurrentDictionary<INamedTypeSymbol, bool>> s_classIsControllerByCompilation =
+            new BoundedCacheWithFactory<Compilation, ConcurrentDictionary<INamedTypeSymbol, bool>>();
+
+        private static readonly ImmutableArray<string> s_dependencyFullTypeNames = ImmutableArray.Create<string>(WellKnownTypeNames.MicrosoftAspNetCoreMvcControllerBase,
```

Comment:
Nit: The source infos are initialized in a static constructor anyway, so could do without a static field here.

Reply:
(The static field being `s_dependencyFullTypeNames`)

Reply:
Same applies to `s_classIsControllerByCompilation`?

Reply:
Nah, the cache is fine.